### DISCUSSION TITLE
Call a function in zustand from another one

### DISCUSSION THREAD

**Discussion by A11yship** [neuefische/questions#132] [Web] [answered]
Hello, 
I write a delete function in zustand. The main function is quite easy 
```
deleteTask(currentTaskId) {
		set(({tasks}) => ({tasks: tasks.filter(task => task.id !== currentTaskId)}));
	},
```
I have nearly the same function a second time, to delete an item from another array. Is it possible to call the second delete function in the one I posted above?
Current [PR](https://github.com/A11yship/capstone-project/pull/28) and [Deployment](https://capstone-project-git-deletetask-a11yship.vercel.app/). Thanks for your help.

**Comment by JessicaLoers** [comment]
Hello - one way is a `if-else statement` which executes the specified `set function`  according the type `currentTasks `or `allTasks`
Perhaps, there's a more efficient way, but this one should work. 


`useStore.js`
``` 
deleteTask(type, id) {
		if (type === 'current') {
			set(({currentTasks}) => ({currentTasks: currentTasks.filter(task => task.id !== id)}));
		} else if (type === 'allTasks') {
			set(({tasks}) => ({tasks: tasks.filter(task => task.id !== id)}));
		}
	}, 
```

```
<Button onClick={() => deleteTask('allTasks', task.id)}>Löschen</Button>
<Button onClick={() => deleteTask('current', task.id)}>Löschen</Button>
```

**Reply by A11yship** [reply]
Thank you for your answer.  I don't want to have one function which deletes ether in the one or other array. If I delete a task from the `allTasks`array I also want to delete it from `currentTasks`. I want to have something like the following:
```
deleteTask(currentTaskId) {
		set(({tasks}) => ({tasks: tasks.filter(task => task.id !== currentTaskId)}));
              deleteFromCurrentTasks(currentTaskId);
	},
```
But this doesn't work. The function isn't found.

**Comment by mpagels** [comment, marked as answer]
Yes, this is possible. You need to provide a `get` parameter when you create the `store` to use it to read from state in actions. Here is the `zustand` example with `get`

```js
import "./styles.css";
import create from "zustand";

const useBearStore = create((set, get) => ({
  bears: 0,
  increasePopulation: () => set((state) => ({ bears: state.bears + 1 })),
  removeAllBears: () => {
    set({ bears: 0 });
    get().increasePopulation();
  }
}));

export default function App() {
  const bears = useBearStore((state) => state.bears);
  const increasePopulation = useBearStore((state) => state.increasePopulation);
  const removeAllBears = useBearStore((state) => state.removeAllBears);

  return (
    <div className="App">
      <h1>{bears} around here ...</h1>
      <button onClick={increasePopulation}>one up</button>
      <button onClick={removeAllBears}>delete</button>
    </div>
  );
}
```

and here is the codesandbox to see how it behaves: https://codesandbox.io/s/young-frost-ckhgmv?file=/src/App.js

Jessicas solution is a good way, too.
